app/controllers/data_requests_controller.rb

```diff
@@ -1,53 +1,34 @@
 class DataRequestsController < ApplicationController
   def index
-    @data_requests = workflow.data_requests.order(created_at: :desc)
+    @data_requests = scope.order(created_at: :desc)
     respond_with @data_requests
   end
 
   def show
-    data_request = workflow.data_requests.find(params[:id])
+    data_request = scope.find(params[:id])
     authorize data_request
     respond_with data_request
   end
 
   def create
-    skip_authorization
-
-    data_request = params[:data_request] || {}
-    requested_data = data_request[:requested_data] || nil
-
-    case requested_data
-    when "extracts"
-      make_request(DataRequest.requested_data[:extracts])
-    when "reductions"
-      make_request(DataRequest.requested_data[:reductions])
-    else
-      head 422
+    DataRequest.transaction do
+      obj = nil
+      args = (params[:data_request] || {}).merge(workflow_id: params[:workflow_id])
+      ctx = {credential: credential}
+
+      data_request = CreatesDataRequests.call(obj, args, ctx)
+      skip_authorization # operations do this themselves and raise if needed
```

Comment:
What Pundit does internally is just set an instance variable to true when you call `authorize`. Long term maybe we want some method in controllers to run an operation, copy that over into the controller, and double check it. For now I've left it at a comment explaining why we skip here.